.github/workflows/ci.yml

```diff
@@ -0,0 +1,65 @@
+# Run basic tests for this app on the latest aiidalab-docker image.
+
+name: continuous-integration
+
+on:
+  [push]
+
+jobs:
+
+  build-docker-image:
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+
+    outputs:
+      image: ${{ steps.tag.outputs.image }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: build image
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          tag_with_ref: true
+          tag_with_sha: true
+          cache_froms: "aiidalab/aiidalab-docker-stack:latest,aiidalab/aiidalab-docker-stack:stable,aiidateam/aiida-core:latest"
```

Comment:
ah, didn't know about this. very nice!

Reply:
Unfortunately it does not work as well as expected. I've investigated ways to improve the performance of docker builds in the context of GitHub actions workflows and my conclusion was that any attempt to introduce "manual caching" is likely not going to lead to any significant performance improvement, but is going to massively increase complexity. The best chance is that GitHub implement layered caching as part of their execution platform, which is likely going to happen at some point, until then we will likely need to live with the current build times.